pkg/controller/plan/kubevirt.go

```diff
@@ -776,12 +778,16 @@ func (r *KubeVirt) virtualMachine(vm *plan.VMStatus) (object *cnv.VirtualMachine
 		originalName = vm.Name
 
 		generatedName := changeVmName(vm.Name, vm.ID)
-		nameExist, errName := r.checkIfVmNameExist(generatedName)
+		nameExist, errName := r.checkIfVmNameExistInNamespace(generatedName, r.Plan.Spec.TargetNamespace)
 		if errName != nil {
 			err = liberr.Wrap(errName)
 			return
 		}
 		if nameExist {
+			// If the name exists and it's at max allowed length, remove 5 chars from the end to not reach the limit after appending vmId
+			if len(generatedName) == NameMaxLength {
+				generatedName = generatedName[0 : NameMaxLength-5]
+			}
 			generatedName = generatedName + "-" + vm.ID[:4]
```

Comment:
that's subjective maybe - but wouldn't it be nicer to append the suffix and then to take a substring from index 5?

Reply:
umm it probably doesn't make much difference but I think it would be better to take it off from the end rather than the beginning, just from the point of the name mining, I think the beginning is more meaningful than the end so butter not cutting it...